src/main/java/io/divolte/server/IncomingRequestProcessor.java

```diff
@@ -150,14 +154,19 @@ public ProcessingDirective process(final DivolteEvent event) {
 
             if (!duplicate || keepDuplicates) {
                 final GenericRecord avroRecord = mapper.newRecordFromExchange(event);
-                final AvroRecordBuffer avroBuffer = AvroRecordBuffer.fromRecord(
+                try {
+                    final AvroRecordBuffer avroBuffer = AvroRecordBuffer.fromRecord(
                         event.partyCookie,
                         event.sessionCookie,
                         event.requestStartTime,
                         event.clientUtcOffset,
                         avroRecord);
-                listener.incomingRequest(event, avroBuffer, avroRecord);
-                doProcess(avroBuffer);
+                    listener.incomingRequest(event, avroBuffer, avroRecord);
+                    doProcess(avroBuffer);
+                } catch (final RuntimeException e) {
```

Comment:
Yeah, I would consider the fromRecord to be part of the mapping (going from event to Avro) and doProcess(...) is part of handling the (then valid) avro record for enqueueing. Any error at that point would really be a bug and not a mapping error.

The listener is really just there for testing purposes (there's certain classes in the HTTP implementation that we cannot mock, so we use this path instead). Running Divolte normally shouldn't use it. If there is an exception in the listener, it's clearly a programming error.